Add customizable link target option to LinkColumn

0xbad0c0d3 opened the pull request:
Allow specifying a target attribute for link opening behavior in LinkColumn. This enables more flexibility in how links are opened, defaulting to "_blank" if no target is provided.

Commit:
Add customizable link target option to LinkColumn

Allow specifying a target attribute for link opening behavior in LinkColumn. This enables more flexibility in how links are opened, defaulting to "_blank" if no target is provided.

WhizzVi commented:
that would be extremely convenient

lukasmasuch commented:
Thanks for the contribution! We have a quite popular feature request for the same feature for `st.link_button`: https://github.com/streamlit/streamlit/issues/7464

And there is an older closed community PR with some related discussions: https://github.com/streamlit/streamlit/pull/8176. Unfortunately, there are some potential security issues related to this feature. Since there is quite some interest in this feature, I will bring it up again in internal discussions.

Commit:
Remove support for _self and _parent link targets

Removed the "_self" and "_parent" options from the LinkTargetType definitions and documentation. The supported link targets are now limited to "_blank" and "_top," simplifying the available options for opening links. And "avoiding" potential issue with iframing arbitrary pages into the main frame

0xbad0c0d3 commented:
What about
```python
import streamlit as st
from streamlit.components.v1 import html

st.write("<a href='something.com' target='_self'>click me</a>", unsafe_allow_html=True)
html("<a href='something.com' target='_self'>click me</a>")
```
looks like this also has security issue?
Or I got the mentioned discussion wrong?

asmeralt commented:
> ... I will bring it up again in internal discussions.

Hi @lukasmasuch , any updates on this? In the PR #8176 I proposed a security measures using HTTP headers. I believe they should mitigate security risks, or am I missing something?
